Dynamic installation of extensions

Armaldio opened the pull request.

Armaldio committed:
add: show window gracefully

Armaldio committed:
download and unzip extension

Armaldio committed:
Merge branch 'develop' into feature/dynamic-extension-installation

Armaldio committed:
add loading spinner

add dynamic extension installation

Armaldio committed:
add dynamic extenison download

add loading page

DannyyyOnGit committed:
Minor Update & Review

TheRealDannyyy commented:
Installation works without manual extensions needed. Disconnect currently doesn't block ads and the webview window shows overflow on start, no clue how to hide it.

Armaldio committed:
removed unused function

Armaldio committed:
Merge branch 'develop' into feature/dynamic-extension-installation

DannyyyOnGit committed:
Design Update

DannyyyOnGit committed:
Speedup Topbar Animation

Fast tool = Fast Animations

TheRealDannyyy reviewed (changes requested):
Installation works without manual extensions needed. I'm not sure how you will handle dynamic installing of updates (not 1st time setup), I assume updating an extension while in use will break it?

Some issues that might need changing:
- Disconnect currently doesn't block ads, workaround needed (see #29)
- Welcome notification should trigger after 1st time setup has finished
- Reload Dialogbox for enable/disabling of extensions (Reload Now; Cancel)
- Fix settings button error after loading screen:
![unknown](https://user-images.githubusercontent.com/15656049/41414802-50524e8c-6fe7-11e8-972a-2d9714455e0e.png)

DannyyyOnGit committed:
Dialogbox Template Designs & Basic About Page

Armaldio committed:
Merge remote-tracking branch 'origin/feature/dynamic-extension-installation' into feature/dynamic-extension-installation

Armaldio committed:
removed console

Armaldio committed:
structure change

add taskbar icon

Armaldio committed:
-

Armaldio committed:
fix ci and moved sounds

DannyyyOnGit committed:
Reinstall NPM

Nothing.

DannyyyOnGit committed:
Settings Menu Update & Folder Cleanup

Features which need to be implemented are currently deactivated.

DannyyyOnGit committed:
Minor Settings Update

Armaldio committed:
updated electron to 3.0

Armaldio committed:
Merge remote-tracking branch 'origin/feature/dynamic-extension-installation' into feature/dynamic-extension-installation

Armaldio committed:
Merge remote-tracking branch 'origin/feature/dynamic-extension-installation' into feature/dynamic-extension-installation

Armaldio committed:
removed unused role

Armaldio committed:
disabled warnings

do not reinstall extensions if already installed

DannyyyOnGit committed:
Filename Changes

Icon name has changed and needs to be changed inside code, if needed.

DannyyyOnGit committed:
Appbar Fix & Name Changes

Armaldio committed:
fixed #35

tempararily disabled extension installation

Armaldio committed:
fixed #23

DannyyyOnGit committed:
Minor Enhancements & New Buttons

DannyyyOnGit committed:
Updated Batch Files

Confirm dialog for re/install and export.

DannyyyOnGit committed:
no newline

DannyyyOnGit committed:
IMPORTANT - Revert To Electron v2.0.3

Fixes broken video and stream playback.

DannyyyOnGit committed:
Dirty CSS Workarounds & Fullscreen Fix

DannyyyOnGit committed:
thx notepad++

DannyyyOnGit committed:
Dialog Update & More "Twitchy" Replacements

"Donate" and "Report Bug" buttons need to open up websites in user's default browser, not TTV Desktop Lite itself.

Currently just placeholders with no function at all.

DannyyyOnGit committed:
CSS Textarea Workaround

DannyyyOnGit committed:
Icon Update

DannyyyOnGit committed:
Fix Export Image Path + More Shadows

Armaldio committed:
fixed #23

Armaldio committed:
fixed #32
update deps
fix weird colors on linux
fix font

Armaldio committed:
show menu only when ready
fix ctrl+r (works only on release mode)
fix listener release so no error anymore on reload
fix for the app trying to reinstall an extension while reload causing an error

Armaldio committed:
fixed stop propagation error

TheRealDannyyy commented:
Will review soon, further changes can be done in the meantime.

DannyyyOnGit committed:
Minor Changes & Fixes

- Fixed all dialog background colors
- Fixed all dialog margins
- Made zooming more precise
- Other smaller changes

DannyyyOnGit committed:
Notepad++ char spacing fix

DannyyyOnGit committed:
Design Update 1

Things aren't complete and there might be CSS that doesn't even have an effect in the code.

DannyyyOnGit committed:
Notepad++ Char Spacing Fix

DannyyyOnGit committed:
Design Update 2

Made things stand out more and added quick access page settings (mostly non-functional).

Armaldio committed:
fix webview positionning
fix window border

Armaldio committed:
fix topbar actions
fix reload/refresh
fix window overflow
add portable target to build
